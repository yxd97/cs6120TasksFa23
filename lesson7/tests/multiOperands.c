#include <stdio.h>
#include <stdlib.h>
// ARGS: 3 5
int main(int argc, char** argv) {
    if (argc < 3) {
        return 1;
    }
    int x1 = atoi(argv[1]);
    int x2 = atoi(argv[2]);
    int r1 = -3 * x1 * x2 ;
    printf("%d x %d x %d = %d\n", -3, x1, x2, r1);
    int r2 = 1233 * x1 * -233;
    printf("%d x %d x %d = %d\n", 1233, x1, -233, r2);
    int r3 = x1 * 213 * x2;
    printf("%d x %d x %d = %d\n", x1, 213, x2, r3);
    int r4 = x2 * -214 * -9;
    printf("%d x %d x %d = %d\n", x2, -214, -9, r4);
    int r5 = x1 * 134 * x2 * 2 * x1 * -3;
    printf("%d x %d x %d x %d x %d x %d = %d\n", x1, 134, x2, 2, x1, -3, r5);
    return 0;
}
