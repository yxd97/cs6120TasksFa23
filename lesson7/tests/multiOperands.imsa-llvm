; ModuleID = 'tests/multiOperands.c'
source_filename = "tests/multiOperands.c"
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-pc-linux-gnu"

@.str = private unnamed_addr constant [19 x i8] c"%d x %d x %d = %d\0A\00", align 1
@.str.1 = private unnamed_addr constant [34 x i8] c"%d x %d x %d x %d x %d x %d = %d\0A\00", align 1

; Function Attrs: noinline nounwind optnone uwtable
define dso_local i32 @main(i32 noundef %0, ptr noundef %1) #0 {
  %3 = alloca i32, align 4
  %4 = alloca i32, align 4
  %5 = alloca ptr, align 8
  %6 = alloca i32, align 4
  %7 = alloca i32, align 4
  %8 = alloca i32, align 4
  %9 = alloca i32, align 4
  %10 = alloca i32, align 4
  %11 = alloca i32, align 4
  %12 = alloca i32, align 4
  store i32 0, ptr %3, align 4
  store i32 %0, ptr %4, align 4
  store ptr %1, ptr %5, align 8
  %13 = load i32, ptr %4, align 4
  %14 = icmp slt i32 %13, 3
  br i1 %14, label %15, label %16

15:                                               ; preds = %2
  store i32 1, ptr %3, align 4
  br label %390

16:                                               ; preds = %2
  %17 = load ptr, ptr %5, align 8
  %18 = getelementptr inbounds ptr, ptr %17, i64 1
  %19 = load ptr, ptr %18, align 8
  %20 = call i32 @atoi(ptr noundef %19) #3
  store i32 %20, ptr %6, align 4
  %21 = load ptr, ptr %5, align 8
  %22 = getelementptr inbounds ptr, ptr %21, i64 2
  %23 = load ptr, ptr %22, align 8
  %24 = call i32 @atoi(ptr noundef %23) #3
  store i32 %24, ptr %7, align 4
  %25 = load i32, ptr %6, align 4
  %26 = mul nsw i32 -3, %25
  %27 = shl i32 %25, 0
  %28 = add i32 0, %27
  %29 = shl i32 %25, 2
  %30 = add i32 %28, %29
  %31 = shl i32 %25, 3
  %32 = add i32 %30, %31
  %33 = shl i32 %25, 4
  %34 = add i32 %32, %33
  %35 = shl i32 %25, 5
  %36 = add i32 %34, %35
  %37 = shl i32 %25, 6
  %38 = add i32 %36, %37
  %39 = shl i32 %25, 7
  %40 = add i32 %38, %39
  %41 = shl i32 %25, 8
  %42 = add i32 %40, %41
  %43 = shl i32 %25, 9
  %44 = add i32 %42, %43
  %45 = shl i32 %25, 10
  %46 = add i32 %44, %45
  %47 = shl i32 %25, 11
  %48 = add i32 %46, %47
  %49 = shl i32 %25, 12
  %50 = add i32 %48, %49
  %51 = shl i32 %25, 13
  %52 = add i32 %50, %51
  %53 = shl i32 %25, 14
  %54 = add i32 %52, %53
  %55 = shl i32 %25, 15
  %56 = add i32 %54, %55
  %57 = shl i32 %25, 16
  %58 = add i32 %56, %57
  %59 = shl i32 %25, 17
  %60 = add i32 %58, %59
  %61 = shl i32 %25, 18
  %62 = add i32 %60, %61
  %63 = shl i32 %25, 19
  %64 = add i32 %62, %63
  %65 = shl i32 %25, 20
  %66 = add i32 %64, %65
  %67 = shl i32 %25, 21
  %68 = add i32 %66, %67
  %69 = shl i32 %25, 22
  %70 = add i32 %68, %69
  %71 = shl i32 %25, 23
  %72 = add i32 %70, %71
  %73 = shl i32 %25, 24
  %74 = add i32 %72, %73
  %75 = shl i32 %25, 25
  %76 = add i32 %74, %75
  %77 = shl i32 %25, 26
  %78 = add i32 %76, %77
  %79 = shl i32 %25, 27
  %80 = add i32 %78, %79
  %81 = shl i32 %25, 28
  %82 = add i32 %80, %81
  %83 = shl i32 %25, 29
  %84 = add i32 %82, %83
  %85 = shl i32 %25, 30
  %86 = add i32 %84, %85
  %87 = shl i32 %25, 31
  %88 = add i32 %86, %87
  %89 = load i32, ptr %7, align 4
  %90 = mul nsw i32 %88, %89
  store i32 %90, ptr %8, align 4
  %91 = load i32, ptr %6, align 4
  %92 = load i32, ptr %7, align 4
  %93 = load i32, ptr %8, align 4
  %94 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef -3, i32 noundef %91, i32 noundef %92, i32 noundef %93)
  %95 = load i32, ptr %6, align 4
  %96 = mul nsw i32 1233, %95
  %97 = shl i32 %95, 0
  %98 = add i32 0, %97
  %99 = shl i32 %95, 4
  %100 = add i32 %98, %99
  %101 = shl i32 %95, 6
  %102 = add i32 %100, %101
  %103 = shl i32 %95, 7
  %104 = add i32 %102, %103
  %105 = shl i32 %95, 10
  %106 = add i32 %104, %105
  %107 = mul nsw i32 %106, -233
  %108 = shl i32 %106, 0
  %109 = add i32 0, %108
  %110 = shl i32 %106, 1
  %111 = add i32 %109, %110
  %112 = shl i32 %106, 2
  %113 = add i32 %111, %112
  %114 = shl i32 %106, 4
  %115 = add i32 %113, %114
  %116 = shl i32 %106, 8
  %117 = add i32 %115, %116
  %118 = shl i32 %106, 9
  %119 = add i32 %117, %118
  %120 = shl i32 %106, 10
  %121 = add i32 %119, %120
  %122 = shl i32 %106, 11
  %123 = add i32 %121, %122
  %124 = shl i32 %106, 12
  %125 = add i32 %123, %124
  %126 = shl i32 %106, 13
  %127 = add i32 %125, %126
  %128 = shl i32 %106, 14
  %129 = add i32 %127, %128
  %130 = shl i32 %106, 15
  %131 = add i32 %129, %130
  %132 = shl i32 %106, 16
  %133 = add i32 %131, %132
  %134 = shl i32 %106, 17
  %135 = add i32 %133, %134
  %136 = shl i32 %106, 18
  %137 = add i32 %135, %136
  %138 = shl i32 %106, 19
  %139 = add i32 %137, %138
  %140 = shl i32 %106, 20
  %141 = add i32 %139, %140
  %142 = shl i32 %106, 21
  %143 = add i32 %141, %142
  %144 = shl i32 %106, 22
  %145 = add i32 %143, %144
  %146 = shl i32 %106, 23
  %147 = add i32 %145, %146
  %148 = shl i32 %106, 24
  %149 = add i32 %147, %148
  %150 = shl i32 %106, 25
  %151 = add i32 %149, %150
  %152 = shl i32 %106, 26
  %153 = add i32 %151, %152
  %154 = shl i32 %106, 27
  %155 = add i32 %153, %154
  %156 = shl i32 %106, 28
  %157 = add i32 %155, %156
  %158 = shl i32 %106, 29
  %159 = add i32 %157, %158
  %160 = shl i32 %106, 30
  %161 = add i32 %159, %160
  %162 = shl i32 %106, 31
  %163 = add i32 %161, %162
  store i32 %163, ptr %9, align 4
  %164 = load i32, ptr %6, align 4
  %165 = load i32, ptr %9, align 4
  %166 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef 1233, i32 noundef %164, i32 noundef -233, i32 noundef %165)
  %167 = load i32, ptr %6, align 4
  %168 = mul nsw i32 %167, 213
  %169 = shl i32 %167, 0
  %170 = add i32 0, %169
  %171 = shl i32 %167, 2
  %172 = add i32 %170, %171
  %173 = shl i32 %167, 4
  %174 = add i32 %172, %173
  %175 = shl i32 %167, 6
  %176 = add i32 %174, %175
  %177 = shl i32 %167, 7
  %178 = add i32 %176, %177
  %179 = load i32, ptr %7, align 4
  %180 = mul nsw i32 %178, %179
  store i32 %180, ptr %10, align 4
  %181 = load i32, ptr %6, align 4
  %182 = load i32, ptr %7, align 4
  %183 = load i32, ptr %10, align 4
  %184 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %181, i32 noundef 213, i32 noundef %182, i32 noundef %183)
  %185 = load i32, ptr %7, align 4
  %186 = mul nsw i32 %185, -214
  %187 = shl i32 %185, 1
  %188 = add i32 0, %187
  %189 = shl i32 %185, 3
  %190 = add i32 %188, %189
  %191 = shl i32 %185, 5
  %192 = add i32 %190, %191
  %193 = shl i32 %185, 8
  %194 = add i32 %192, %193
  %195 = shl i32 %185, 9
  %196 = add i32 %194, %195
  %197 = shl i32 %185, 10
  %198 = add i32 %196, %197
  %199 = shl i32 %185, 11
  %200 = add i32 %198, %199
  %201 = shl i32 %185, 12
  %202 = add i32 %200, %201
  %203 = shl i32 %185, 13
  %204 = add i32 %202, %203
  %205 = shl i32 %185, 14
  %206 = add i32 %204, %205
  %207 = shl i32 %185, 15
  %208 = add i32 %206, %207
  %209 = shl i32 %185, 16
  %210 = add i32 %208, %209
  %211 = shl i32 %185, 17
  %212 = add i32 %210, %211
  %213 = shl i32 %185, 18
  %214 = add i32 %212, %213
  %215 = shl i32 %185, 19
  %216 = add i32 %214, %215
  %217 = shl i32 %185, 20
  %218 = add i32 %216, %217
  %219 = shl i32 %185, 21
  %220 = add i32 %218, %219
  %221 = shl i32 %185, 22
  %222 = add i32 %220, %221
  %223 = shl i32 %185, 23
  %224 = add i32 %222, %223
  %225 = shl i32 %185, 24
  %226 = add i32 %224, %225
  %227 = shl i32 %185, 25
  %228 = add i32 %226, %227
  %229 = shl i32 %185, 26
  %230 = add i32 %228, %229
  %231 = shl i32 %185, 27
  %232 = add i32 %230, %231
  %233 = shl i32 %185, 28
  %234 = add i32 %232, %233
  %235 = shl i32 %185, 29
  %236 = add i32 %234, %235
  %237 = shl i32 %185, 30
  %238 = add i32 %236, %237
  %239 = shl i32 %185, 31
  %240 = add i32 %238, %239
  %241 = mul nsw i32 %240, -9
  %242 = shl i32 %240, 0
  %243 = add i32 0, %242
  %244 = shl i32 %240, 1
  %245 = add i32 %243, %244
  %246 = shl i32 %240, 2
  %247 = add i32 %245, %246
  %248 = shl i32 %240, 4
  %249 = add i32 %247, %248
  %250 = shl i32 %240, 5
  %251 = add i32 %249, %250
  %252 = shl i32 %240, 6
  %253 = add i32 %251, %252
  %254 = shl i32 %240, 7
  %255 = add i32 %253, %254
  %256 = shl i32 %240, 8
  %257 = add i32 %255, %256
  %258 = shl i32 %240, 9
  %259 = add i32 %257, %258
  %260 = shl i32 %240, 10
  %261 = add i32 %259, %260
  %262 = shl i32 %240, 11
  %263 = add i32 %261, %262
  %264 = shl i32 %240, 12
  %265 = add i32 %263, %264
  %266 = shl i32 %240, 13
  %267 = add i32 %265, %266
  %268 = shl i32 %240, 14
  %269 = add i32 %267, %268
  %270 = shl i32 %240, 15
  %271 = add i32 %269, %270
  %272 = shl i32 %240, 16
  %273 = add i32 %271, %272
  %274 = shl i32 %240, 17
  %275 = add i32 %273, %274
  %276 = shl i32 %240, 18
  %277 = add i32 %275, %276
  %278 = shl i32 %240, 19
  %279 = add i32 %277, %278
  %280 = shl i32 %240, 20
  %281 = add i32 %279, %280
  %282 = shl i32 %240, 21
  %283 = add i32 %281, %282
  %284 = shl i32 %240, 22
  %285 = add i32 %283, %284
  %286 = shl i32 %240, 23
  %287 = add i32 %285, %286
  %288 = shl i32 %240, 24
  %289 = add i32 %287, %288
  %290 = shl i32 %240, 25
  %291 = add i32 %289, %290
  %292 = shl i32 %240, 26
  %293 = add i32 %291, %292
  %294 = shl i32 %240, 27
  %295 = add i32 %293, %294
  %296 = shl i32 %240, 28
  %297 = add i32 %295, %296
  %298 = shl i32 %240, 29
  %299 = add i32 %297, %298
  %300 = shl i32 %240, 30
  %301 = add i32 %299, %300
  %302 = shl i32 %240, 31
  %303 = add i32 %301, %302
  store i32 %303, ptr %11, align 4
  %304 = load i32, ptr %7, align 4
  %305 = load i32, ptr %11, align 4
  %306 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %304, i32 noundef -214, i32 noundef -9, i32 noundef %305)
  %307 = load i32, ptr %6, align 4
  %308 = mul nsw i32 %307, 134
  %309 = shl i32 %307, 1
  %310 = add i32 0, %309
  %311 = shl i32 %307, 2
  %312 = add i32 %310, %311
  %313 = shl i32 %307, 7
  %314 = add i32 %312, %313
  %315 = load i32, ptr %7, align 4
  %316 = mul nsw i32 %314, %315
  %317 = mul nsw i32 %316, 2
  %318 = shl i32 %316, 1
  %319 = add i32 0, %318
  %320 = load i32, ptr %6, align 4
  %321 = mul nsw i32 %319, %320
  %322 = mul nsw i32 %321, -3
  %323 = shl i32 %321, 0
  %324 = add i32 0, %323
  %325 = shl i32 %321, 2
  %326 = add i32 %324, %325
  %327 = shl i32 %321, 3
  %328 = add i32 %326, %327
  %329 = shl i32 %321, 4
  %330 = add i32 %328, %329
  %331 = shl i32 %321, 5
  %332 = add i32 %330, %331
  %333 = shl i32 %321, 6
  %334 = add i32 %332, %333
  %335 = shl i32 %321, 7
  %336 = add i32 %334, %335
  %337 = shl i32 %321, 8
  %338 = add i32 %336, %337
  %339 = shl i32 %321, 9
  %340 = add i32 %338, %339
  %341 = shl i32 %321, 10
  %342 = add i32 %340, %341
  %343 = shl i32 %321, 11
  %344 = add i32 %342, %343
  %345 = shl i32 %321, 12
  %346 = add i32 %344, %345
  %347 = shl i32 %321, 13
  %348 = add i32 %346, %347
  %349 = shl i32 %321, 14
  %350 = add i32 %348, %349
  %351 = shl i32 %321, 15
  %352 = add i32 %350, %351
  %353 = shl i32 %321, 16
  %354 = add i32 %352, %353
  %355 = shl i32 %321, 17
  %356 = add i32 %354, %355
  %357 = shl i32 %321, 18
  %358 = add i32 %356, %357
  %359 = shl i32 %321, 19
  %360 = add i32 %358, %359
  %361 = shl i32 %321, 20
  %362 = add i32 %360, %361
  %363 = shl i32 %321, 21
  %364 = add i32 %362, %363
  %365 = shl i32 %321, 22
  %366 = add i32 %364, %365
  %367 = shl i32 %321, 23
  %368 = add i32 %366, %367
  %369 = shl i32 %321, 24
  %370 = add i32 %368, %369
  %371 = shl i32 %321, 25
  %372 = add i32 %370, %371
  %373 = shl i32 %321, 26
  %374 = add i32 %372, %373
  %375 = shl i32 %321, 27
  %376 = add i32 %374, %375
  %377 = shl i32 %321, 28
  %378 = add i32 %376, %377
  %379 = shl i32 %321, 29
  %380 = add i32 %378, %379
  %381 = shl i32 %321, 30
  %382 = add i32 %380, %381
  %383 = shl i32 %321, 31
  %384 = add i32 %382, %383
  store i32 %384, ptr %12, align 4
  %385 = load i32, ptr %6, align 4
  %386 = load i32, ptr %7, align 4
  %387 = load i32, ptr %6, align 4
  %388 = load i32, ptr %12, align 4
  %389 = call i32 (ptr, ...) @printf(ptr noundef @.str.1, i32 noundef %385, i32 noundef 134, i32 noundef %386, i32 noundef 2, i32 noundef %387, i32 noundef -3, i32 noundef %388)
  store i32 0, ptr %3, align 4
  br label %390

390:                                              ; preds = %16, %15
  %391 = load i32, ptr %3, align 4
  ret i32 %391
}

; Function Attrs: nounwind willreturn memory(read)
declare i32 @atoi(ptr noundef) #1

declare i32 @printf(ptr noundef, ...) #2

attributes #0 = { noinline nounwind optnone uwtable "frame-pointer"="all" "min-legal-vector-width"="0" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #1 = { nounwind willreturn memory(read) "frame-pointer"="all" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #2 = { "frame-pointer"="all" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #3 = { nounwind willreturn memory(read) }

!llvm.module.flags = !{!0, !1, !2, !3, !4}
!llvm.ident = !{!5}

!0 = !{i32 1, !"wchar_size", i32 4}
!1 = !{i32 8, !"PIC Level", i32 2}
!2 = !{i32 7, !"PIE Level", i32 2}
!3 = !{i32 7, !"uwtable", i32 2}
!4 = !{i32 7, !"frame-pointer", i32 2}
!5 = !{!"Ubuntu clang version 17.0.2 (++20230929073225+098e653a5bed-1~exp1~20230929073241.48)"}
