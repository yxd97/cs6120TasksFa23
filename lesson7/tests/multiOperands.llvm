; ModuleID = 'tests/multiOperands.c'
source_filename = "tests/multiOperands.c"
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-pc-linux-gnu"

@.str = private unnamed_addr constant [19 x i8] c"%d x %d x %d = %d\0A\00", align 1
@.str.1 = private unnamed_addr constant [34 x i8] c"%d x %d x %d x %d x %d x %d = %d\0A\00", align 1

; Function Attrs: noinline nounwind optnone uwtable
define dso_local i32 @main(i32 noundef %0, ptr noundef %1) #0 {
  %3 = alloca i32, align 4
  %4 = alloca i32, align 4
  %5 = alloca ptr, align 8
  %6 = alloca i32, align 4
  %7 = alloca i32, align 4
  %8 = alloca i32, align 4
  %9 = alloca i32, align 4
  %10 = alloca i32, align 4
  %11 = alloca i32, align 4
  %12 = alloca i32, align 4
  store i32 0, ptr %3, align 4
  store i32 %0, ptr %4, align 4
  store ptr %1, ptr %5, align 8
  %13 = load i32, ptr %4, align 4
  %14 = icmp slt i32 %13, 3
  br i1 %14, label %15, label %16

15:                                               ; preds = %2
  store i32 1, ptr %3, align 4
  br label %66

16:                                               ; preds = %2
  %17 = load ptr, ptr %5, align 8
  %18 = getelementptr inbounds ptr, ptr %17, i64 1
  %19 = load ptr, ptr %18, align 8
  %20 = call i32 @atoi(ptr noundef %19) #3
  store i32 %20, ptr %6, align 4
  %21 = load ptr, ptr %5, align 8
  %22 = getelementptr inbounds ptr, ptr %21, i64 2
  %23 = load ptr, ptr %22, align 8
  %24 = call i32 @atoi(ptr noundef %23) #3
  store i32 %24, ptr %7, align 4
  %25 = load i32, ptr %6, align 4
  %26 = mul nsw i32 -3, %25
  %27 = load i32, ptr %7, align 4
  %28 = mul nsw i32 %26, %27
  store i32 %28, ptr %8, align 4
  %29 = load i32, ptr %6, align 4
  %30 = load i32, ptr %7, align 4
  %31 = load i32, ptr %8, align 4
  %32 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef -3, i32 noundef %29, i32 noundef %30, i32 noundef %31)
  %33 = load i32, ptr %6, align 4
  %34 = mul nsw i32 1233, %33
  %35 = mul nsw i32 %34, -233
  store i32 %35, ptr %9, align 4
  %36 = load i32, ptr %6, align 4
  %37 = load i32, ptr %9, align 4
  %38 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef 1233, i32 noundef %36, i32 noundef -233, i32 noundef %37)
  %39 = load i32, ptr %6, align 4
  %40 = mul nsw i32 %39, 213
  %41 = load i32, ptr %7, align 4
  %42 = mul nsw i32 %40, %41
  store i32 %42, ptr %10, align 4
  %43 = load i32, ptr %6, align 4
  %44 = load i32, ptr %7, align 4
  %45 = load i32, ptr %10, align 4
  %46 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %43, i32 noundef 213, i32 noundef %44, i32 noundef %45)
  %47 = load i32, ptr %7, align 4
  %48 = mul nsw i32 %47, -214
  %49 = mul nsw i32 %48, -9
  store i32 %49, ptr %11, align 4
  %50 = load i32, ptr %7, align 4
  %51 = load i32, ptr %11, align 4
  %52 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %50, i32 noundef -214, i32 noundef -9, i32 noundef %51)
  %53 = load i32, ptr %6, align 4
  %54 = mul nsw i32 %53, 134
  %55 = load i32, ptr %7, align 4
  %56 = mul nsw i32 %54, %55
  %57 = mul nsw i32 %56, 2
  %58 = load i32, ptr %6, align 4
  %59 = mul nsw i32 %57, %58
  %60 = mul nsw i32 %59, -3
  store i32 %60, ptr %12, align 4
  %61 = load i32, ptr %6, align 4
  %62 = load i32, ptr %7, align 4
  %63 = load i32, ptr %6, align 4
  %64 = load i32, ptr %12, align 4
  %65 = call i32 (ptr, ...) @printf(ptr noundef @.str.1, i32 noundef %61, i32 noundef 134, i32 noundef %62, i32 noundef 2, i32 noundef %63, i32 noundef -3, i32 noundef %64)
  store i32 0, ptr %3, align 4
  br label %66

66:                                               ; preds = %16, %15
  %67 = load i32, ptr %3, align 4
  ret i32 %67
}

; Function Attrs: nounwind willreturn memory(read)
declare i32 @atoi(ptr noundef) #1

declare i32 @printf(ptr noundef, ...) #2

attributes #0 = { noinline nounwind optnone uwtable "frame-pointer"="all" "min-legal-vector-width"="0" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #1 = { nounwind willreturn memory(read) "frame-pointer"="all" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #2 = { "frame-pointer"="all" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #3 = { nounwind willreturn memory(read) }

!llvm.module.flags = !{!0, !1, !2, !3, !4}
!llvm.ident = !{!5}

!0 = !{i32 1, !"wchar_size", i32 4}
!1 = !{i32 8, !"PIC Level", i32 2}
!2 = !{i32 7, !"PIE Level", i32 2}
!3 = !{i32 7, !"uwtable", i32 2}
!4 = !{i32 7, !"frame-pointer", i32 2}
!5 = !{!"Ubuntu clang version 17.0.2 (++20230929073225+098e653a5bed-1~exp1~20230929073241.48)"}
