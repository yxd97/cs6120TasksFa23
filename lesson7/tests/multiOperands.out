-3 x 3 x 5 = -45
1233 x 3 x -233 = -861867
3 x 213 x 5 = 3195
5 x -214 x -9 = 9630
3 x 134 x 5 x 2 x 3 x -3 = -36180
